src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -69,9 +75,9 @@ public Model(ModelMetadata modelMetadata, @Nullable byte[] modelBlob) {
      * @param modelBlob binary representation of model template index. Can be null if model is not yet in CREATED state.
      * @param modelID model identifier
      */
-    public Model(ModelMetadata modelMetadata, @Nullable byte[] modelBlob, String modelID) {
-        this(modelMetadata,modelBlob);
-        this.modelID = Objects.requireNonNull(modelID, "model id must not be null");
+    public Model(ModelMetadata modelMetadata, @Nullable byte[] modelBlob, @NonNull String modelID) {
```

Comment:
NonNull doesnt provide runtime validation does it? Why remove requireNonNull?

Reply:
Ack